hw4/03-form.js

```diff
@@ -1,16 +1,62 @@
-const express = require('express');
+const http = require("http");
+const express = require("express");
+const parser = require("body-parser");
+let fs = require("fs");
+const port = process.env.PORT || 5001;
+const routes = ["form", "submit"];
+
+let getRoutes = () => {
+  //Build a list of links for the root index.
+  let result = "";
+  routes.forEach(
+    (elem) => (result += `<li><a href="/${elem}">${elem}</a></li>`)
+  );
+  return result;
+};
+
 const app = express();
-const port = process.env.PORT || 5000;
+app.use(parser.urlencoded({ extended: true }));
+
+app.get("/", (req, res) => {
+  res.writeHead(302, { Location: "/form" });
+  res.end(); //Response complete.
+});
+
+app.get("/form", (req, res) => {
+  //res.set('Content-Type', "text/html");
+  fs.readFile("03-form.html", "utf8", (err, data) => {
```

Comment:
Hm, I wasn't able to get this to work out-of-the-box. When I grabbed the repo it didn't come with an `03-form.html` so I got `Whoops! File not found!` when I tried going to `localhost:5001/form`.

I guessed that it was the same 03-form.html from your hw3, so I copied that and pasted it into the root directory, as it looked like it was pulling from there, and that did get it to load. However, once it loaded I noticed a few errors that make me think you probably made changes to hw4 version of 03-form.html, so this version just isn't working right for me.